### DISCUSSION TITLE
Point forecasts ingestion?

### DISCUSSION THREAD

**Discussion by Chris-J-Steele** [dtcenter/METplus#1532] [Use Cases] [answered]
Hello,

This question might have a very quick answer. Is it possible for MET to ingest point forecasts, rather than those on a grid? I can see that there's Point2Grid, which I guess could be adapted in some way.

The forecast files are CF-Compliant NetCDF, but consist of a list of points.

cheers

Chris

**Comment by hsoh-u** [comment]
Yes, point2grid supports three NetCDF files and the python embedding:
1. MET generated point observation data (ascii2nc, madis2nc, pb2nc, lidas2nc, and  ioda2nc)
2. GOES 16 and 17 data
3. CF-Compliant NetCDF with three coordinate variables & the data variable.

Three coordinate variables  at the CF-Compliant NetCDF:
- time variable should meet at least one of following conditions:
  1. The variable name is "T"
  2. the standard_name attribute is "t", "time", "Time", "TIME", "datetime", "Datetime", or "DATETIME"
  3. the units attribute is "(<time_unit> since <reference_time>"
    - <time_unit> is second, seconds, minute, minutes, hour, hours, day, or days
    - <reference_time> is "YYYY-mm-DDTHH:MM:SS" or "YYYY-mm-DD HH:MM:SS"
- latitude variable should meet at least one of following conditions:
  1. the standard_name attribute is "lat", "LAT", "Lat", "Latitude", "latitude", or  "LATITUDE"
  2. the units attribute is "degrees_north", "degree_north", "degree_N", "degrees_N", "degreeN", or "degreesN"
  3. the _CoordinateAxisType attribute is "x" or "X" 
- longitude variable should meet at least one of following conditions:
  1. the standard_name attribute is "lon", "LON", "Lon",  "Longitude", "longitude", or "LONGITUDE"
  2. the units attribute is "degrees_east", "degree_east", "degree_E", "degrees_E",  "degreeE", or "degreesE"
  3. the _CoordinateAxisType attribute is "y" or "Y"

**Reply by Chris-J-Steele** [reply]
That's interesting thanks. Is there no other way then to use point forecasts without putting it onto a grid?

**Comment by CPKalb** [comment, marked as answer]
Hi Chris:

What are you wanting to do with the point forecasts?  If you are wanting to verify them against point observations, then the best way is to match up these forecast to observations and put them in MET's matched pair format (MPR) which can then be run through stat_analysis (https://met.readthedocs.io/en/latest/Users_Guide/stat-analysis.html).  They can be ingested into stat_analysis through python embedding (as a python list in MPR format) or as MPR .stat files.  I have previously done this and can give examples if it is what you are looking to do.  Additionally, information about the matched pair line can be found near the end of the point_stat section of the users guide.

Thanks,
Christina

**Reply by Chris-J-Steele** [reply]
Hi Christina,

Thanks very much for responding. At this stage, I wanted to check whether it was possible in MET as I couldn't find anything in the documentation to suggest that it was. It does make sense to both match and reformat the data for use with stat-analysis. I'll propose that we try this out and shout if I get stuck

thanks again

Chris

**Reply by CPKalb** [reply]
Hi Chris:

MET doesn't have any support for matching point forecasts to point observations at this time.  But, you can compute statistics on point forecasts by matching them and then running the matched data through stat_analysis.  Let me know if you have other questions.

Christina
